### PR TITLE
Create initial TPDM base models 

### PR DESCRIPTION
**Pull request by zaharenka**

### CODE REVIEW

File: models/staging/tpdm/base/base_tpdm__post_secondary_institutions.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        -- descriptors
+        {{ extract_descriptor('v:postSecondaryInstitutionLevelDescriptor::string')}} as post_secondary_institution_level,
+        {{ extract_descriptor('v:administrativeFundingControlDescriptor::string')}}  as administrative_funding_control,
+        {{ extract_descriptor('v:federalLocaleCodeDescriptor::string')}}             as federal_locale_code,
+        -- unflattened lists
+        v:mediumOfInstructions as v_medium_of_instructions
+    from student_academic_records
+)
+
+select * from renamed
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__post_secondary_institutions.sql]: please end sql files with a newline

```suggestion
select * from renamed

```

File: models/staging/tpdm/base/base_tpdm__survey_response_person_target_associations.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        v:id::string                                               as record_guid,
+        v:surveyResponseReference:surveyResponseIdentifier::string as survey_response_id,
+        v:personReference:personId:string                          as person_id,
+        -- references 
+        v:surveyResponseReference as survey_response_reference,
+        v:personReference as person_reference
+    from survey_response_person_target_associations
+)
+
+select * from renamed
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__survey_response_person_target_associations.sql]: please end sql files with a newline
```suggestion
select * from renamed

```

File: models/staging/tpdm/base/base_tpdm__education_organizations.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        v:webSite::string                as web_site,
+        -- descriptors
+        {{ extract_descriptor('v:operationalStatusDescriptor::string') }} as operational_tatus,
+        -- unnested lists
+        v:identificationCodes as identification_codes,
+        v:categories as categories,
+        v:addresses as addresses,
+        v:internationalAddresses as international_addresses,
+        v:institutionTelephones as institution_telephones,
+        v:indicators as indicators
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__education_organizations.sql]: could the `as` renames line 22-27 be indented to the same level as 17-18?

File: models/staging/tpdm/base/base_tpdm__student_academic_records.sql [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        file_row_number,
+        filename,
+        is_deleted,
+        v:id::string                                                  as record_guid,
+
+        v:studentReference:studentUniqueId::string                    as student_unique_id,
+        v:educationOrganizationReference:educationOrganizationId::int as ed_org_id,
+        v:educationOrganizationReference:link:rel::string             as ed_org_type,
+        v:schoolYearTypeReference:schoolYear::int                     as school_year,
+        v:projectedGraduationDate::date                          as projected_graduation_date,
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__student_academic_records.sql]: could the indentation be lined up here?

File: models/staging/tpdm/base/base_tpdm__candidate_educator_preparation_program_associations.sql [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+),
+renamed as (
+    select 
+        tenant_code,
+        api_year,
+        pull_timestamp,
+        last_modified_timestamp,
+        file_row_number,
+        filename,
+        is_deleted,
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__candidate_educator_preparation_program_associations.sql:12]: would it make sense to define the type for these attributes as well?
For example `is_deleted` should be a boolean.

/cc @holdorph thoughts?

File: models/staging/tpdm/base/base_tpdm__candidate_educator_preparation_program_associations.sql [hunk 6]
```diff
@@ -0,0 +1,35 @@
+with candidate_educator_preparation_program_associations as (
+    {{ source_edfi3('candidate_educator_preparation_program_associations') }}
+),
+renamed as (
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__candidate_educator_preparation_program_associations.sql:4]: could we add a newline between CTEs?

```suggestion
),

renamed as (
```

File: models/staging/tpdm/base/base_tpdm__education_organizations.sql [hunk 7]
```diff
@@ -0,0 +1,31 @@
+with education_organizations as (
+    {{ source_edfi3('education_organizations') }}
+),
+renamed as (
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__education_organizations.sql:4]: could we add a newline between CTEs?
```suggestion
),

renamed as (
```

File: models/staging/tpdm/base/base_tpdm__persons.sql [hunk 8]
```diff
@@ -0,0 +1,20 @@
+with persons as (
+    {{ source_edfi3('persons') }}
+),
+renamed as (
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__persons.sql:4]: could we add a newline between CTEs?
```suggestion
),

renamed as (
```

File: models/staging/tpdm/base/base_tpdm__post_secondary_institutions.sql [hunk 9]
```diff
@@ -0,0 +1,25 @@
+with post_secondary_institutions as (
+    {{ source_edfi3('post_secondary_institutions') }}
+),
+renamed as (
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__post_secondary_institutions.sql:4]: could we add a newline between CTEs?
```suggestion
),

renamed as (
```

File: models/staging/tpdm/base/base_tpdm__schools.sql [hunk 10]
```diff
@@ -0,0 +1,37 @@
+with schools as (
+    {{ source_edfi3('schools') }}
+),
+renamed as (
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__schools.sql:4]: could we add a newline between CTEs?
```suggestion
),

renamed as (
```

File: models/staging/tpdm/base/base_tpdm__survey_responses.sql [hunk 11]
```diff
@@ -0,0 +1,33 @@
+with survey_responses as (
+    {{ source_edfi3('survey_responses') }}
+),
+renamed as (
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__survey_responses.sql:4]: could we add a newline between CTEs?
```suggestion
),

renamed as (
```

File: models/staging/tpdm/base/base_tpdm__survey_section_responses.sql [hunk 12]
```diff
@@ -0,0 +1,26 @@
+with base_tpdm__survey_section_responses as (
+    {{ source_edfi3('base_tpdm__survey_section_responses') }}
+),
+renamed as (
```
Comment (ChristianMurphy) [inline models/staging/tpdm/base/base_tpdm__survey_section_responses.sql:4]: could we add a newline between CTEs?
```suggestion
),

renamed as (
```

File: models/staging/tpdm/base/base_tpdm__candidate_educator_preparation_program_associations.sql (hunk 6 above)
Comment (maxalmazov) [reply models/staging/tpdm/base/base_tpdm__candidate_educator_preparation_program_associations.sql:4]: Hi @ChristianMurphy, it is good idea it looks better with new line, but there is not new line in other models in current repository, if we add new line we break general code style. If it is ok, we will add new line

**Comment by maxalmazov** [comment]
Sorry it was closed due to `feature/tpdm` branch was deleted by me, I created [the same pull request](https://github.com/Unicon/edu_edfi_source/pull/5) with `main` as a target

File: models/staging/tpdm/base/base_tpdm__candidate_educator_preparation_program_associations.sql (hunk 6 above)
Comment (ChristianMurphy) [reply models/staging/tpdm/base/base_tpdm__candidate_educator_preparation_program_associations.sql:4]: Both styles are in the existing code

https://github.com/Unicon/edu_edfi_source/blob/c865d2839664e34670a6d73148b7ca9a1b21babe/models/staging/edfi_3/intermediate/int_ef3__deduped_descriptors.sql#L1-L29

**Pull request closed without merging**